flight/search/plot_algo.py

```diff
@@ -0,0 +1,601 @@
+    search area, exponentially reducing the time the algorithm takes
+
+    Methods
+    -------
+    analyze_cell(i : int, j: int, s: int, cell_map: CellMap) -> int
+        Returns the number of in-bounds cells
+    get_circum_square_r(r: int) -> int
+        returns the side length of the square inside the view radius circle
+    __init_compressed_grid(cell_size : int, cell_map : CellMap) -> NDArray:
+        Returns empty compressed grid with correct dimensions
+    compress(radius: int, cell_map: CellMap) -> list[list[tuple[bool, int, int]]]
+        compresses the given cell_map into larger cells based on the given radius
+    """
+
+    @staticmethod
+    def analyze_cell(i: int, j: int, size: int, cell_map: CellMap) -> int:
+        """
+        Given the compressed index, returns the number of valid locations
+        within the cell
+
+        Parameters
+        ----------
+        i : int
+            the row of the compressed cell
+        j : int
+            the column of the compressed cell
+        size: int
+            size of the cell
+        cell_map : CellMap
+            the map being compressed
+
+        Returns
+        -------
+        value : int
+            the number of valid cells within the compressed cell
+        """
+        score: int = 0
+        row_start: int = size * i
+        row_end: int = min((size * (i + 1) - 1), len(cell_map.data) - 1)
+        col_start: int = size * j
+        col_end: int = min((size * (j + 1) - 1), len(cell_map[0]) - 1)
+
+        for row in range(row_start, row_end + 1):
+            for col in range(col_start, col_end + 1):
+                try:
+                    if cell_map[row][col].is_valid:
+                        score += 1
+                except IndexError:
+                    pass
+
+        return score
+
+    @staticmethod
+    def get_circum_square_r(radius: int) -> int:
+        """
+        Given the view radius of the seeker, return the radius of the
+        circumscribed square that lies within it
+
+        Parameters
+        ----------
+        radius : int
+            Radius of the circle
+
+        Returns
+        -------
+        side : int
+            side length of the square
+        """
+
+        return radius  # max(floor(sqrt(r/2)), 1)
+
+    @staticmethod
+    def __init_compressed_grid(cell_size: int, cell_map: CellMap) -> NDArray[int8]:
+        """
+        Returns an empty grid for the compressed map
+
+        Parameters
+        ----------
+        cell_size : int
+            the side length of each square cell
+        cell_map : CellMap
+            the map being compressed
+
+        Returns
+        -------
+        empty_grid : NDArray
+            an empty grid of three points repres
+        """
+        cols: int = floor(len(cell_map[0]) / cell_size)
+        rows: int = floor(len(cell_map.data) / cell_size)
+        return zeros((rows, cols), dtype=int8)
```

Comment:
When you change to `import numpy as np`, you will have to change this to `np.zeros((rows, cols), dtype=np.int8)`. Same thing anywhere either of these are used throughout the file